### ISSUE TITLE
Title on top bar in upload View is shown sometimes owncloud instead of Uploads

### ISSUE THREAD

**Issue by nasli** [owncloud/android#1524]
### Expected behaviour
The title does not change 

### Actual behaviour
Uploads title change to "ownCloud" and we are still on the uploads view

### Steps to reproduce
1. Tap on left button menu and select uploads
2. Tap on the left button menu
3. You do not select any option in the lateral menu and came back to uploads view

### Environment 
Nexus 5, v5

**Comment by LukeOwncloud** [comment]
Note: Uploads view show menu button at top left corner. The other views (settings and protocol) show a left arrow. 

If also in the uploads view an arrow was shown, this issue would probably be resolved?! Plus: Consistent behavior within the app.

**Comment by LukeOwncloud** [comment]
Note 2 : It can even happen that a photo filename is displayed as title. I am not quite sure how to reproduce. I was clicking on current, failed, and completed upload items. Then menu button and back, as described by nasli. Then title was IMG_....jpg.

**Comment by nasli** [comment]
I rather left this behaviour for all the views, and only change to the files list view if the user tap over Files in the menu. 

About Note2, we are aware of that but same here, thanks for checking @LukeOwncloud !

**Comment by AndyScherzinger** [comment]
At the moment I would ague to go for the arrow since you basically switch to another "context". 

In the future via @tobiasKaminsky work we will also be able to have filters in the drawer like "offline files" where you stay in the same context. So you will still have the drawer but that is another thing. Having the drawer on all screens has been discussed between @jancborchardt and me and we decided to stick with the arrow.

Hope that helps :)

**Comment by nasli** [comment]
@LukeOwncloud , note2 catch out! https://github.com/owncloud/android/issues/1530

**Comment by malkomich** [comment]
@AndyScherzinger the arrow style in the action bar implies that you want to go back in the navigation, like when you back to a parent folder for instance.
The burguer menu is wished to be used If the user wants to change the context of the application going to a different view like in this case.
Also, the arrow is being used for setting view because this view is a special one in Android which have a different inheritance flow, but for the rest of views the burguer menu would be more meaningful for a navigation bar.

**Comment by AndyScherzinger** [comment]
@malkomich that is true, the question is what is more expected by the user, to navigate somewhere else, or to navigate back/up simply because flow wise navigating somewhere else even if it is the main file list adds to the navigational stack for back navigation. Both things work imho but I am not an UX expert as to what is the users expectation here. @jancborchardt any input here? :smiley:

**Comment by nasli** [comment]
As an user, I would like a menu, and go home only if I tap on it

**Comment by LukeOwncloud** [comment]
Works for me now. :+1:

**Comment by davivel** [comment]
As already said, the "Google pattern" prefers the burger menu for any activity that offers navigation drawer and appears as a top-level option in it. Settings view is an exception to the pattern (for us and for any app with it, check yourself), mainly because settings activities tipically inherit from http://developer.android.com/intl/es/reference/android/preference/PreferenceActivity.html , while activities with navigation drawer inherit from http://developer.android.com/intl/es/reference/android/support/v7/app/AppCompatActivity.html, and Java doesn't support multiple inheritance.

The "logs" activity ( @LukeOwncloud , you mean that with "protocol", right? ) should not be taken as a reference, since it was built just-for-developer-eyes , and is not specially careful with design guidelines.

**Comment by davivel** [comment]
> Note 2 : It can even happen that a photo filename is displayed as title

I think that's #1530, should be fixed now too.

**Comment by AndyScherzinger** [comment]
> The "logs" activity ( @LukeOwncloud , you mean that with "protocol", right? ) should not be taken as a reference, since it was built just-for-developer-eyes , and is not specially careful with design guidelines.
Exactly, that one should stay that way especially since #1432 will lead to the logs item being moved to the settings like version number etc. and than the back/up arrow would be the right thing again :smiley:

**Comment by nasli** [comment]
:+1: We can close this issue

**Comment by jancborchardt** [comment]
@AndyScherzinger just to clarify what the expectation would be:

- If I’m in »All files«, the title says »ownCloud« and there’s the menu icon on the top left
- If I’m in »On device« or »Uploads« the title should reflect that of course! Also, ideally there should be the menu icon on the top left triggering the menu. From a user perspective, these are not entirely different »views«, it’s just a filter on the same stuff. So the rest of the interface should behave the same!
- If I’m in »Settings« the title should of course be »Settings« as well. It’s fine to have the back button since it’s not a file list, but in an ideal world where I don’t care about the technical and platform conventions, the settings view would be in the content area exactly like the other ones. ;)

TL;DR: Even though the content changes, the interface like top bar (except the right side tools), sidebar etc should stay consistent in some way.

**Comment by AndyScherzinger** [comment]
@jancborchardt sounds perfectly fine to me :) Besides this is imho how it is implemented at the moment is it not?
